epics-service/src/main/java/edu/gemini/epics/impl/ReadOnlyEpicsChannelImpl.java

```diff
@@ -25,21 +26,38 @@
  *         Date: 11/7/11
  */
 public class ReadOnlyEpicsChannelImpl<T> implements ReadOnlyClientEpicsChannel<T> {
+    private static final Logger LOG = Logger.getLogger(ReadOnlyEpicsChannelImpl.class.getName());
     protected final CAJChannel channel;
     protected double timeout;
+    protected int retries;
     private final Map<EpicsListener<?>, MonitorListenerPair> listeners = new HashMap<EpicsListener<?>, MonitorListenerPair>();
 
-    public ReadOnlyEpicsChannelImpl(CAJChannel channel, double timeout) {
+    public ReadOnlyEpicsChannelImpl(CAJChannel channel, double timeout, int retries) {
         this.channel = channel;
         this.timeout =  timeout;
+        this.retries = retries;
     }
 
     @Override
     public DBR getDBR() throws CAException, TimeoutException {
-        DBR dbr;
-        synchronized (channel.getContext()) {
-            dbr = channel.get();
-            channel.getContext().pendIO(timeout);
+        DBR dbr = null;
+        boolean success = false;
+        for(int i = 0; !success && i<=retries; i++) {
+            try {
+                synchronized (channel.getContext()) {
```

Comment:
I wonder if this is the right place for the `lock`. Should it be outside the for/retry loop?

Reply:
I am not even sure the `synchronized` is even needed. The `get()` is supposed to be an asynchronous operation, thus the need for calling `pendIO()` later.

Reply:
Right, it maybe because we are `synchronizing` in other places? Once you do sync on a value in one place you should do the same everywhere that value is used.

But if it is not needed we may just as well drop it